### PR TITLE
Add a settings page!

### PR DESCRIPTION
**Pull request by ashika-verma**
Added a stopwatch button on the least used station in the MTA. This button toggles whether or not a train's time is displayed in minutes (default time visualization) or in seconds.

image

image

### CODE REVIEW

**Comment by jamespfennell** [comment]
Thank you for the pull request!

Is this idea of putting the stopwatch in the Beach 105 st station that most users won't want to tweak it, and so it's enough to have it as an Easter egg feature?

I do think that seconds-level granularity for trains is not really more informative than minutes-level granularity because transit data is not that accurate. (I think the MTA may even officially say something like this.) But I think it's in the spirit of the app being open source that users can configure it that way if they want. Rather than an Easter egg, I think the "right" way to do this would be to have a settings page in the app in which this could be toggled. What do you think? Feel free to say you don't have the bandwidth for such a larger change. :)

Also, just want to note that I'm going traveling for 3 weeks without a computer so I may not be responsive to GitHub issues!

**Comment by ashika-verma** [comment]
Hey! Thanks for the thoughtful response! 

I do agree that the seconds level granularity is probably not accurate, but I'm frequently quite curious what's going on in the backend while I'm waiting at the platform. I left it in for now (mostly to demo the settings page) but I'd be totally okay removing it later!

Here's the settings page:
image

File: src/hooks/seconds.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useLocalStorage } from "./localstorage";
+
+export const useSeconds = () => {
+  const [useSeconds, setUseSeconds] = useLocalStorage("rtr.seconds", false);
+
+  return {
+    toggleSeconds: () => setUseSeconds(!useSeconds),
+    getUseSeconds: () => !!useSeconds,
```
Comment (bporter816) [inline src/hooks/seconds.ts]: Might not need the `!!` if you have the default of `false`

**Comment by jamespfennell** [comment]
Thanks for the settings page!! It looks great!

I just have one last question before submitting. In the new code we use local storage to store a Boolean indicating whether seconds are used or not. Instead of storing a single Boolean, would it be difficult to store a json struct, with one Boolean field right now for this setting? I think going forward it would make it easier to add new settings by just adding new fields to this struct and not needing to add more local storage code.

Feel free to say no and we can submit as is :)

Btw just a reminder that I’m traveling without a computer and even if we merge this PR I won’t be able to deploy it until I get home in mid January. Deploying new versions currently requires SSHing into a Digital Ocean VM which I can only do on my laptop…

**Comment by jamespfennell** [comment]
Thanks for all the work here!! Sorry for the late response - just recently backup from a trip and catching up.

I'm going to merge this now.

Just one note that I was thinking that perhaps the settings page isn't important enough yet to be in the top bar, especially because there's not much in there right now. I like the idea of keeping the top bar very small to focus on my main "user journeys". In a follow up commit I may tweak the top bar accordingly.

**Pull request merged**